### PR TITLE
chore: update tough cookie

### LINKED ISSUE #27261: Vulnerability Issue reported for tough-cookie, latest version of @cypress/request is dependent of tough-cookie@2.5.0
### Current behavior

It's working fine but having vulnerability issue on package @cypress/request or others consuming tough-cookie@2.5.0 which is <4.1.3

### Desired behavior

Upgrade tough-cookie version to tough-cookie@4.1.3 for all packages having dependencies on this

Here is an example from latest tag, yarn-lock

"@cypress/request@^2.88.11":
  version "2.88.11"
  resolved "https://registry.yarnpkg.com/@cypress/request/-/request-2.88.11.tgz#5a4c7399bc2d7e7ed56e92ce5acb620c8b187047"
  integrity sha512-M83/wfQ1EkspjkE2lNWNV5ui2Cv7UCv1swW1DqljahbzLVWltcsexQh8jYtuS/vzFXP+HySntGM83ZXA9fn17w==
  dependencies:
    aws-sign2 "~0.7.0"
    aws4 "^1.8.0"
    caseless "~0.12.0"
    combined-stream "~1.0.6"
    extend "~3.0.2"
    forever-agent "~0.6.1"
    form-data "~2.3.2"
    http-signature "~1.3.6"
    is-typedarray "~1.0.0"
    isstream "~0.1.2"
    json-stringify-safe "~5.0.1"
    mime-types "~2.1.19"
    performance-now "^2.1.0"
    qs "~6.10.3"
    safe-buffer "^5.1.2"
    **tough-cookie "~2.5.0"**
    tunnel-agent "^0.6.0"
    uuid "^8.3.2"

### Test code to reproduce

NA

### Cypress Version

12.17.1

### Node version

v16.15.0

### Operating System

windows

### Debug Logs

```shell
NA
```

### Other

_No response_

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes #27261

### Additional details
- This patches `@cypress/request`, `@cypress/request-promise` and `start-server-and-test` which have dependencies on a vulnerable version of `tough-cookie`. It DOES leave ONE instance of `tough-cookie` left at `2.5` in `packages/icons`. This one will be difficult to patch since it is a peer dep of `@packages#icons#to-ico#resize-img#jimp#request` and  `to-ico` hasn't been updated in over 5 years

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/web-config/node-jsdom-setup.ts
Comment (jordanpowell88) [inline packages/web-config/node-jsdom-setup.ts:1]: The version of `jsdom` we were using was using `tough-cookie` `2.5` but after further investigation this was the only file using `jsdom` and that it wasn't anywhere in the monorepo. I don't think we need this anymore

File: package.json [hunk 1]
```diff
@@ -198,7 +198,6 @@
     "shelljs": "0.8.5",
     "sinon": "7.3.2",
     "snap-shot-it": "7.9.3",
-    "start-server-and-test": "1.10.8",
```
Comment (jordanpowell88) [inline package.json:201]: `start-server-and-test` version `1` depends upon `tough-cookie` `2.5` but I don't see any use of `start-server-and-test` in the repo. I think we can remove this

File: patches/whatwg-url+7.1.0.dev.patch
Comment (jordanpowell88) [inline patches/whatwg-url+7.1.0.dev.patch:1]: We were getting [this error in CI](https://app.circleci.com/pipelines/github/cypress-io/cypress/55393/workflows/6ea304f5-1f25-4f40-adcd-9092ea01c01e/jobs/2292778) saying that we don't need this patch anymore. Removing it appears to be the fix.

Comment (AtofStryker) [reply patches/whatwg-url+7.1.0.dev.patch:1]: Guess we figured out what was updating the hoisted dep. Thank you for removing this patch! We needed it for the webpack v5 update and glad it was shortlived.

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ _Released 08/15/2023 (PENDING)_
...
+- Upgraded [`tough-cookie`](https://www.npmjs.com/package/tough-cookie) from `2.5` to `4.1.3`, [`@cypress/request`](https://www.npmjs.com/package/@cypress/request) from `2.88.11` to `2.88.12` and [`@cypress/request-promise`](https://www.npmjs.com/package/@cypress/request-promise) from `4.2.6` to `4.2.7` to address a [security vulnerability](https://security.snyk.io/vuln/SNYK-JS-TOUGHCOOKIE-5672873). Fixed in [#27515](https://github.com/cypress-io/cypress/pull/27515)
```
Comment (AtofStryker) [inline cli/CHANGELOG.md]: I think there were some nested deps on 2.5, but I think in the repo itself we went from `4.0.0` to `4.1.3`.
```suggestion
- Upgraded [`tough-cookie`](https://www.npmjs.com/package/tough-cookie) from `4.0.0` to `4.1.3`, [`@cypress/request`](https://www.npmjs.com/package/@cypress/request) from `2.88.11` to `2.88.12` and [`@cypress/request-promise`](https://www.npmjs.com/package/@cypress/request-promise) from `4.2.6` to `4.2.7` to address a [security vulnerability](https://security.snyk.io/vuln/SNYK-JS-TOUGHCOOKIE-5672873). Fixed in [#27515](https://github.com/cypress-io/cypress/pull/27515)
```

Comment (jordanpowell88) [reply cli/CHANGELOG.md]: Good point. Should be resolved now

**Review by AtofStryker — APPROVED** [review]

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**